src/charm.py

```diff
@@ -541,6 +545,16 @@ def _has_required_fields(self, conf_or_rel: _ConfigOrRelation) -> bool:
         field_names = [f'_{f.replace("-", "_")}' for f in REQUIRED_INGRESS_RELATION_FIELDS]
         return all(getattr(conf_or_rel, f) for f in field_names)
 
+    def _invalid_hostname_check(self, hostname: str) -> Optional[re.Match]:
+        """Check if the hostname is valid according to RFC 1123.
+
+        Args:
+            hostname: Ingress hostname
+        """
+        return re.match(
+            "[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*", hostname
```

Comment:
I assume this came from `RFC 1123`, would it be worth including a comment with a link to where the regex is defined?